p4-16/psa/PSA.mdk

```diff
@@ -911,36 +897,40 @@ the cloned copy to `clone_port`.
+
+The ingress parser distinguishes the resubmitted packet from the
+original packet with the `packet_path` field in
+`ingress_parser_intrinsic_metadata_t`. The ingress parser can choose a
+different algorithm to parse the resubmitted packet. Similarly, the
+ingress pipeline can choose to process the resubmitted packet with
+different actions as opposed to the ones used to process the original
 packet.
 
-The ingress parser is able to distinguish between the original packet
-and the resubmitted packet using the `packet_path` field in
-`ingress_parser_intrinsic_metadata_t`. The ingress parser is allowed
-to choose a different algorithm to parse the resubmitted packet.
-Similarly, the ingress pipeline can choose to process the resubmitted
-packet with different actions as opposed to the ones used to process
-the original packet.
-
-PSA specifies that the resubmit operation can only be used in the ingress pipeline. The egress
-pipeline cannot resubmit packets. As described in Section [#packet-paths], there is no mandated
-mechanism in PSA to prevent a single received packet from creating packets that continue to
-recirculate, resubmit, or clone from egress to egress indefinitely. However, targets may impose
-limits on the number of resubmissions, recirculations, or clones.
-
-Packet resubmission is intended to increase the capacity and
- flexibility of the packet processing pipeline. For example, because
- the same packet is processed by the ingress pipeline multiple times,
- it effectively increase the amount of operations on the packet by N
- folds, where N is the number of times the packet is resubmitted.
- Another use case is deploy multiple packet processing algorithms on
- the same packet. For example, the original packet can be parsed,
- modified and resubmitted in the first pass. Then, the resubmitted
- packet can be parsed and modified using a different algorithm.
+PSA specifies that the resubmit operation can only be used in the
+ingress pipeline. The egress pipeline cannot resubmit packets. As
+described in Section [#packet-paths], there is no mandated mechanism
+in PSA to prevent a single received packet from creating packets that
+continue to recirculate, resubmit, or clone from egress to egress
+indefinitely. However, targets may impose limits on the number of
+resubmissions, recirculations, or clones.
+
+One use case of packet resubmission is to increase the capacity and
+flexibility of the packet processing pipeline. For example, because
+the same packet is processed by the ingress pipeline multiple times,
+it effectively increase the amount of operations on the packet by N
+folds, where N is the number of times the packet is resubmitted.
+
+Another use case is to deploy multiple packet processing algorithms on
+the same packet. For example, the original packet can be parsed,
+modified and resubmitted in the first pass. Then, the resubmitted
+packet can be parsed and modified using a different algorithm.
```

Comment:
It seems a little bit misleading to say that the original packet can be "modified" here, doesn't it?  I mean, I know it can modify the packet before resubmitting it, but all such modifications will be lost, true?